app/Jobs/TranscodeVideo.php

```diff
@@ -149,13 +154,12 @@ protected function transcodeVideo(): void
         // Set the necessary file path information.
         $this->setFilePaths();
 
-
         // Generate MP4.
         $this->generateMp4($video);
         // Generate HLS
-        $this->saveVideo(StreamingFormat::HLS->configure($video), StreamingFormat::HLS->value);
+        $this->saveVideo(StreamingFormat::HLS->configure($video, $this->encoder), StreamingFormat::HLS->value);
```

Comment:
why dont we pass the enum case and the video to automatically resolve these within the saveVideo-method? this also reduces the amount of duplicate calls

Reply:
done